packages/tracing/src/hubextensions.ts

```diff
@@ -206,15 +208,66 @@ export function startIdleTransaction(
  * @private
  */
 export function _addTracingExtensions(): void {
+  // FIXME: This is problematic, because `getMainCarrier` always return a valid Carrier, but because
+  // of how we set the types, we need to perform an unnecessary check here.
   const carrier = getMainCarrier();
-  if (carrier.__SENTRY__) {
-    carrier.__SENTRY__.extensions = carrier.__SENTRY__.extensions || {};
-    if (!carrier.__SENTRY__.extensions.startTransaction) {
-      carrier.__SENTRY__.extensions.startTransaction = _startTransaction;
-    }
-    if (!carrier.__SENTRY__.extensions.traceHeaders) {
-      carrier.__SENTRY__.extensions.traceHeaders = traceHeaders;
-    }
+  if (!carrier.__SENTRY__) {
+    return;
+  }
+  carrier.__SENTRY__.extensions = carrier.__SENTRY__.extensions || {};
+  if (!carrier.__SENTRY__.extensions.startTransaction) {
+    carrier.__SENTRY__.extensions.startTransaction = _startTransaction;
+  }
+  if (!carrier.__SENTRY__.extensions.traceHeaders) {
+    carrier.__SENTRY__.extensions.traceHeaders = traceHeaders;
+  }
+}
```

Comment:
I failed to see how this is related to the database integrations work; maybe if it is not it would be best in a separate PR to avoid confusion.

Reply:
It's not related, it's a simple refactor to eager return. It doesn't change any logic of the code, so I didn't create a new PR for it.